pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -194,27 +205,29 @@ func getDesiredBootstrapDataForBackup(
 }
 
 func getDesiredDiskDataForBackup(
-	ctx goctx.Context,
-	vcVM *object.VirtualMachine,
+	opts BackupOptions,
 	ecMap map[string]string) (string, error) {
-	deviceList, err := vcVM.Device(ctx)
+	// Return an empty string to skip backup if no disk uuid to PVC is specified.
+	if len(opts.DiskUUIDToPVC) == 0 {
+		return "", nil
+	}
+
+	deviceList, err := opts.VcVM.Device(opts.VMCtx)
 	if err != nil {
 		return "", err
 	}
 
 	var diskData []VMDiskData
 	for _, device := range deviceList.SelectByType((*types.VirtualDisk)(nil)) {
 		if disk, ok := device.(*types.VirtualDisk); ok {
-			vmDiskData := VMDiskData{}
-			if disk.VDiskId != nil {
-				vmDiskData.VDiskID = disk.VDiskId.Id
-			}
 			if b, ok := disk.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
-				vmDiskData.FileName = b.FileName
-			}
-			// Only add the disk data if it's not empty.
-			if vmDiskData != (VMDiskData{}) {
-				diskData = append(diskData, vmDiskData)
+				if pvc, ok := opts.DiskUUIDToPVC[b.Uuid]; ok {
```

Comment:
Can you confirm that the `uuid` field of the disk will always be set for FCDs?
The documentation of the backing doesn't mention anything (in fact, this field was introduced in API version 2.5, whereas FCDs were introduced only around 6.7 IIRC.

https://vdc-download.vmware.com/vmwb-repository/dcr-public/790263bc-bd30-48f1-af12-ed36055d718b/e5f17bfc-ecba-40bf-a04f-376bbb11e811/vim.vm.device.VirtualDisk.FlatVer2BackingInfo.html#field_detail

```
uuid* | xsd:string | Disk UUID for the virtual disk, if available.  Since VI API 2.5
```

Same question for v1a2.

Reply:
Discussed offline and confirmed that this field will be set when the corresponding `CnsNodeVmAttachment` is reconciled, after which it will be populated to `VM.Status.Volumes`.